ecl/hql/hqlgram2.cpp

```diff
@@ -11958,16 +11985,15 @@ IHqlExpression *HqlGram::yyParse(bool _parsingTemplateAttribute, bool catchAbort
     }
     catch (IException * e)
     {
-        if (e->errorCode() != ERR_ABORT_PARSING)
+        if (catchAbort || e->errorCode() != ERR_ABORT_PARSING)
```

Comment:
I agree that combining the two looks a lot neater. I did explain this on Thursday. Firstly there are two separate error codes, one for an abort (2000) and the other for too many errors (2092). Since these are error codes, there is really no need to explicitly state an error code to state that an error has occurred. The reason, however, that it (parsing aborted) is there, is due to it throwing and thus throwing a (that) specific message. I changed the above code because the throw is in some places caught and reported as an error via the handle but then wouldn't be without this change. So it would in the cases that the throw isn't caught earlier in the parser code and is thus caught by eclcc code and not in others. I thought it would be confusing to sometimes see that abort message and sometimes not.
Another point is that ERR_ABORT_PARSING was used as a test for existing messages on whether to report or continue throwing. Please advise of a preferred change.